packages/pluggableWidgets/switch-web/src/Switch.tsx

```diff
@@ -0,0 +1,55 @@
+import { createElement, FunctionComponent, KeyboardEvent, MouseEvent, useCallback } from "react";
+import { executeAction, isAvailable } from "@mendix/piw-utils-internal";
+import { ValueStatus } from "mendix";
+import { Switch as SwitchComponent } from "./components/Switch";
+
+import { SwitchContainerProps } from "../typings/SwitchProps";
+import "./ui/switch-main.scss";
+
+export const Switch: FunctionComponent<SwitchContainerProps> = props => {
+    const isChecked = isAvailable(props.booleanAttribute);
+    const editable = !props.booleanAttribute.readOnly;
+
+    function invokeActionAndMaybeToggleValue() {
+        if (props.booleanAttribute.status === ValueStatus.Available) {
+            props.booleanAttribute.setValue(!props.booleanAttribute.value);
+        }
+        executeAction(props.action);
+    }
+
+    const onClick = useCallback(
+        (event: MouseEvent<HTMLDivElement>) => {
+            event.preventDefault();
+            if (editable) {
+                invokeActionAndMaybeToggleValue();
```

Comment:
If you make this function a useCallback defining its own dependencies (separation of concerns), then you don't need to add the dependencies for this method in line 27 nor 36

Reply:
I agree, good one 👍